Utils/azuretools-core/src/com/microsoft/azuretools/telemetry/TelemetryParameter.java

```diff
@@ -0,0 +1,145 @@
+        TelemetryParameter RUN = new TelemetryParameter(MODULE, "run-function-app");
+        TelemetryParameter CREATE_TRIGGER = new TelemetryParameter(MODULE, "create-function-trigger");
+        TelemetryParameter CREATE_PROJECT = new TelemetryParameter(MODULE, "create-function-project");
+        TelemetryParameter CREATE = new TelemetryParameter(MODULE, "create-function-app");
+        TelemetryParameter DEPLOY = new TelemetryParameter(MODULE, "deploy-function-app");
+        TelemetryParameter DELETE = new TelemetryParameter(MODULE, "delete-function-app");
+        TelemetryParameter START = new TelemetryParameter(MODULE, "start-function-app");
+        TelemetryParameter STOP = new TelemetryParameter(MODULE, "stop-function-app");
+        TelemetryParameter RESTART = new TelemetryParameter(MODULE, "restart-function-app");
+        TelemetryParameter SHOW_PROPERTIES = new TelemetryParameter(MODULE, "showprop-function-app");
+        TelemetryParameter START_STREAMING_LOG = new TelemetryParameter(MODULE, "start-streaming-log-function-app");
+        TelemetryParameter STOP_STREAMING_LOG = new TelemetryParameter(MODULE, "stop-streaming-log-function-app");
+        TelemetryParameter OPEN_IN_PORTAL = new TelemetryParameter(MODULE, "open-inbrowser-function-app");
+    }
+
+    public interface MySQL {
+        String MODULE = "azure-mysql";
+
+        TelemetryParameter CREATE = new TelemetryParameter(MODULE, buildOperationName(MODULE, "create"));
+        TelemetryParameter DELETE = new TelemetryParameter(MODULE, buildOperationName(MODULE, "delete"));
+        TelemetryParameter START = new TelemetryParameter(MODULE, buildOperationName(MODULE, "start"));
+        TelemetryParameter STOP = new TelemetryParameter(MODULE, buildOperationName(MODULE, "stop"));
+        TelemetryParameter RESTART = new TelemetryParameter(MODULE, buildOperationName(MODULE, "restart"));
+        TelemetryParameter OPEN_IN_PORTAL = new TelemetryParameter(MODULE, buildOperationName(MODULE, "open-in-portal"));
+        TelemetryParameter SHOW_PROPERTIES = new TelemetryParameter(MODULE, buildOperationName(MODULE, "show-properties"));
+        TelemetryParameter CONNECT_TO_SERVER = new TelemetryParameter(MODULE, buildOperationName(MODULE, "connect-to-server"));
+        TelemetryParameter BIND_INTO = new TelemetryParameter(MODULE, buildOperationName(MODULE, "bind-into"));
+    }
+
+    @Override
+    public int hashCode() {
+        return this.operationName.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        } else if (!(obj instanceof TelemetryParameter)) {
+            return false;
+        } else if (obj == this) {
+            return true;
+        } else {
+            TelemetryParameter rhs = (TelemetryParameter) obj;
+            return this.operationName.equalsIgnoreCase(rhs.operationName);
```

Comment:
use EqualsBuilder?

Reply:
equals is enough here. just need to compare String and the constructor is private for this class.